presto-main/src/main/java/com/facebook/presto/type/FloatOperators.java

```diff
@@ -17,25 +17,86 @@
 import com.facebook.presto.spi.PrestoException;
 import com.facebook.presto.spi.type.StandardTypes;
 
+import static com.facebook.presto.metadata.OperatorType.ADD;
+import static com.facebook.presto.metadata.OperatorType.DIVIDE;
+import static com.facebook.presto.metadata.OperatorType.MODULUS;
+import static com.facebook.presto.metadata.OperatorType.MULTIPLY;
 import static com.facebook.presto.metadata.OperatorType.NEGATION;
+import static com.facebook.presto.metadata.OperatorType.SUBTRACT;
 import static com.facebook.presto.spi.StandardErrorCode.NUMERIC_VALUE_OUT_OF_RANGE;
+import static java.lang.String.format;
 
 public final class FloatOperators
 {
     private FloatOperators()
     {
     }
 
+    @ScalarOperator(ADD)
+    @SqlType(StandardTypes.FLOAT)
+    public static long add(@SqlType(StandardTypes.FLOAT) long left, @SqlType(StandardTypes.FLOAT) long right)
+    {
+        ensureFloatInRange(left);
```

Comment:
Adding two floats cannot produce not-float (in opposite to integer overflowing to bigint). In worst case it'll produce `infinity` but will never get out of range, so I assume no boundaries check are needed (?).